File: android/src/main/java/expo/modules/settings/ExpoSettingsModule.kt
package expo.modules.settings

import expo.modules.kotlin.modules.modules
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoSettingsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoSettings")

    Function("getTheme") {
      return@Function "system"
    }
  }
}

File: ios/ExpoSettingsModule.swift
import ExpoModulesCore

public class ExpoSettingsModule: Module {
    public func definition() -> ModuleDefinition {
        Name("ExpoSettings")
        
        Events("onChangeTheme")
          
        Function("setTheme") { (theme: String) -> Void in
            UserDefaults.standard.set(theme, forKey: "theme")
            sendEvent("onChangeTheme", ["theme" : theme])
            print("Theme set to \(theme)") 
        }

        Function("getTheme") { () -> String in
            UserDefaults.standard.string(forKey: "theme") ??
            "system"
        }
    }
}
